### ISSUE TITLE
[Go] Implement Usage of Golang arenas (go1.20+)

### ISSUE THREAD

**Issue by ivanmemruk** [apache/arrow-go#62]
### Describe the enhancement requested

Golang v1.20 has added an experimental `arenas` package, which allows more "manual" memory allocation and could be potentially useful for the golang arrow library.

I am opening this issue to see if there has been any interest or any attempt to try to use `arenas` for arrow. If there hasn't, I could potentially try working on that in the nearest few months and contribute back.

### Component(s)

Go

**Comment by zeroshade** [comment]
Closing this one out. Three years on, the concern @spenczar raised in https://github.com/apache/arrow-go/issues/62#issuecomment-2319657465 turned out to be the deciding factor: arenas never graduated out of `GOEXPERIMENT=arenas`, the proposal (golang/go#51317) remains on hold, and the Go team's guidance is still that the API may change or be removed without warning.

Building an arena-backed `memory.Allocator` into arrow-go would mean shipping a public surface gated on an experiment that we can't promise will exist in the next Go release, so I don't think it's something we should take on speculatively.

If arenas are ever unfrozen with a committed API, this is absolutely worth revisiting — an arena `memory.Allocator` remains the natural first integration point, and the benchmarks in the repo are still the right way to evaluate it. Happy to reopen at that point.

Thanks @ivanmemruk for raising it and @spenczar for the early heads-up.
